bind.go

```diff
@@ -125,12 +125,30 @@ func (b *DefaultBinder) Bind(i interface{}, c Context) (err error) {
 
 // bindData will bind data ONLY fields in destination struct that have EXPLICIT tag
 func (b *DefaultBinder) bindData(destination interface{}, data map[string][]string, tag string) error {
-	if destination == nil || len(data) == 0 {
+	if destination == nil {
 		return nil
 	}
 	typ := reflect.TypeOf(destination).Elem()
 	val := reflect.ValueOf(destination).Elem()
 
+	if len(data) == 0 {
```

Comment:
This does not default cases when some fields are given and some missing 

See testcases:
```go
func TestBindQueryParamsWithDefaultValue(t *testing.T) {
	type person struct {
		Name string `query:"name" default:"bob"`
		Age  int    `query:"age" default:"18"`
	}

	var testCases = []struct {
		name   string
		when   string
		expect person
	}{
		{
			name:   "ok, all fields filled",
			when:   `/?name=foo&age=1`,
			expect: person{Name: "foo", Age: 1},
		},
		{
			name:   "ok, none of the fields filled",
			when:   `/`,
			expect: person{Name: "bob", Age: 18},
		},
		{
			name:   "ok, single field filled",
			when:   `/name=xxx`,
			expect: person{Name: "xxx", Age: 18},
		},
	}
	for _, tc := range testCases {
		t.Run(tc.name, func(t *testing.T) {
			e := New()
			req := httptest.NewRequest(http.MethodGet, tc.when, nil)
			rec := httptest.NewRecorder()
			c := e.NewContext(req, rec)
			p := person{}
			err := c.Bind(&p)
			if assert.NoError(t, err) {
				assert.Equal(t, tc.expect, p)
			}
		})
	}
}
```